instructor/client_cohere.py

```diff
@@ -0,0 +1,37 @@
+import cohere
+import instructor
+
+from typing import overload
+
+
+@overload
+def from_cohere(
+    client: cohere.Client,
```

Comment:
is there an async client we need to watch out for?

Reply:
oh yes there's [`cohere.AsyncClient`](https://github.com/cohere-ai/cohere-python/blob/c7d5d5ea8451d35ea054187e24356b99cb484c88/src/cohere/client.py#L248)

for some reason `instructor.utils.is_async` is not recognizing its `client.chat` function as async and so patching doesn't work...will send a follow up a bit later today

Reply:
Should support both now!